File: 0155-min-stack/0155-min-stack.java

class MinStack {

    Stack<Pair<Integer, Integer>> stack;
    public MinStack() {
        stack = new Stack<>();
    }
    
    public void push(int val) {
        if (!stack.isEmpty()) {
            Pair<Integer, Integer> curr = new Pair<>
            (val, Math.min(val, stack.peek().getValue()));
            stack.push(curr);
        } else {
            Pair<Integer, Integer> curr = new Pair<>(val, val);
            stack.push(curr);
        }
        return;

    }
    
    public void pop() {
        stack.pop();
        return;
    }
    
    public int top() {
        return stack.peek().getKey();
    }
    
    public int getMin() {
        return stack.peek().getValue();
    }
}

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack obj = new MinStack();
 * obj.push(val);
 * obj.pop();
 * int param_3 = obj.top();
 * int param_4 = obj.getMin();
 */

File: 0155-min-stack/0155-min-stack.cpp
class MinStack {
stack<pair<int, int>> minStack;
public:
    MinStack() {
        
    }
    
    void push(int value) {
        if (minStack.empty()) {
            minStack.push(
                pair(value, value)
            );
        } else {
            minStack.push(
                pair(value, min(value, minStack.top().second))
            );
        }
    }
    
    void pop() {
        minStack.pop();
    }
    
    int top() {
        return minStack.top().first;
    }
    
    int getMin() {
        return minStack.top().second;
    }
};

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack* obj = new MinStack();
 * obj->push(value);
 * obj->pop();
 * int param_3 = obj->top();
 * int param_4 = obj->getMin();
 */